### ISSUE TITLE
Document deployment

### ISSUE THREAD

**Issue by lovelydinosaur** [Kludex/uvicorn#145]
Example configurations with gunicorn/supervisor/circus, and examples of nginx fronting the service.

Jotting down some examples for now...

Nginx proxy headers, for use with `--proxy-headers`:

```
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Proto $proxy_x_forwarded_proto;
proxy_set_header X-Forwarded-Port $proxy_x_forwarded_port;
```

**supervisord.conf**:

```
[supervisord]

[fcgi-program:uvicorn]
socket=tcp://localhost:9002
command=venv/bin/uvicorn --fd 0 example:App
numprocs=2
process_name=uvicorn-%(process_num)d

# Just for debugging
stdout_logfile=/dev/stdout
stdout_logfile_maxbytes=0
stderr_logfile=/dev/stderr
stderr_logfile_maxbytes=0
```

Then test it with `supervisord -n` (No demonize)

---

**circus.ini**

```
[watcher:web]
cmd = venv/bin/uvicorn --fd $(circus.sockets.web) example:App
use_sockets = True
numprocesses = 2

[env:web]
LC_ALL=en_US.UTF-8
LANG=en_US.UTF-8

[socket:web]
host = 0.0.0.0
port = 8000
```

Then run with `circusd circus.ini`

**Comment by lovelydinosaur** [comment]
See https://github.com/encode/uvicorn/pull/164

**Comment by lovelydinosaur** [comment]
Closed via #164. See https://www.uvicorn.org/
